### DISCUSSION TITLE
Wine limited to 1 CPU core on M1 MBA

### DISCUSSION THREAD

**Discussion by hiimkir** [The-Wineskin-Project/WineskinServer#303] [Q&A] [answered]
Hi, I'm new to macOS, so I don't really understand how things work here. For some reason Wineskin Advanced shows "limited to 1 CPU core (set at an OS level for all applications)", which I believe to be the reason for very poor performance (3-4 fps in 2003 game Knights of the Old Republic). I'm on WS11WCX64b21.2.0 with 2.9.1.1 wrapper. Task Manager shows only 59 available threads (out of >2k available in macOS) and 700MB of RAM.

**Comment by Gcenx** [comment, marked as answer]
> limited to 1 CPU core (set at an OS level for all applications)

Ignore this checkbox it’s grayed out and even if you somehow forced it to be enabled it wouldn’t work as that only worked on 10.8.

> Knights of the Old Republic

You’ve said 3-4fps so that makes me think your on an Apple Silicon Mac as I’ve found many 32Bit games that ran decently on my Intel Mac became a slideshow on the M1 mini.

This isn’t to say this will affects all 32Bit games.

Also a major difference with Apple Silicon systems is it doesn’t natively support OpenGL instead everything runs on-top of the Metal API, the only reason Apple didn’t optimize this OpenGL emulation is it’s used for Safari. OpenGL has been long officially depreciated by Apple on macOS.

**Reply by hiimkir** [reply]
Yeah, I've finally noticed this in my logs:
```
wined3d_dll_init Setting multithreaded command stream to 0x1
wined3d_dll_init Disabling 3D support
```
So I guess this is an OpenGL issue.

But! I've also been using some of code testing utilities by my university (they only come in 32bit windows executables) and they also work very slow compared to my old Intel laptop. As far as I checked these programs do not use any GPU functionality, so this must be a CPU/RAM issue

**Reply by Gcenx** [reply]
> But! I've also been using some of code testing utilities by my university (they only come in 32bit windows executables) and they also work very slow compared to my old Intel laptop. As far as I checked this programs do not use any GPU functionality, so this must be a CPU/RAM issue

Rosetta2 is known to not work well with some optimization levels/floating point operations. It was designed to work nicely with the defaults that Apple recommends for native applications/games.

I’ve been building wine using the same optimizations as CodeWeavers so there’s really nothing else that can be done on my end unfortunately.
